osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/GridMeasurementPublishedEventConverter.java

```diff
@@ -36,30 +45,84 @@
 
     @Override
     public GridMeasurementPublishedEvent convert(final String source,
             final Type<? extends GridMeasurementPublishedEvent> destinationType, final MappingContext mappingContext) {
 
-        StringArrayToAnalogList stringArrayToAnalogList = null;
-        final String[] values = source.split(";");
-        if (values.length == SIMPLE_END_INDEX) {
-            stringArrayToAnalogList = new SimpleStringToAnalogList();
-        } else if (values.length == LsPeakShavingMeasurementType.getNumberOfElements() + 1) {
-            stringArrayToAnalogList = new LsMeasurementMessageToAnalogList();
-        } else {
-            LOGGER.error("String '{}' does not have the expected amount of fields, abandoning conversion", source);
+        LsMeasurementMessageToAnalogList stringArrayToAnalogList = null;
+        LOGGER.info("Source string: {}", source);
+
+        try {
+            final Payload payload = this.objectMapper.readValue(source, Payload[].class)[0];
+
+            final String[] values = (payload.gisnr + ", " + String.join(", ", payload.data)).split(", ");
+            LOGGER.info("Values length: {} and values: {}", values.length, Arrays.toString(values));
+
+            if (values.length == LsPeakShavingMeasurementType.getNumberOfElements() + 1) {
+                stringArrayToAnalogList = new LsMeasurementMessageToAnalogList();
+            } else {
+                LOGGER.error("String '{}' does not have the expected amount of fields, abandoning conversion", source);
+                return null;
+            }
+
+            final String eanCode = values[0];
+            final PowerSystemResource powerSystemResource = new PowerSystemResource(eanCode,
+                    UUID.randomUUID().toString(), new ArrayList<Name>());
+            final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-mm-dd HH:mm:ss");
+            final Date date = dateFormat.parse(payload.date);
+            final long createdDateTime = date.getTime();
+            LOGGER.info("CreatedDateTime: {}", createdDateTime);
+            return new GridMeasurementPublishedEvent(createdDateTime, eanCode, UUID.randomUUID().toString(),
+                    "GridMeasurementPublishedEvent", new ArrayList<Name>(), powerSystemResource,
+                    stringArrayToAnalogList.convertToAnalogList(values));
+        } catch (final JsonMappingException e) {
+            LOGGER.error("Caught an error mapping a JSON string to Payload. {}", source, e);
+            return null;
+        } catch (final JsonProcessingException e) {
+            LOGGER.error("Caught an error processing a JSON string to Payload. {}", source, e);
+            return null;
+        } catch (final ParseException e) {
+            LOGGER.error("Date could not be parsed corrrectly. Date format is: yyyy-mm-dd HH:mm:ss, "
+                    + "however the provided date was not the correct format. {}", source, e);
             return null;
```

Comment:
In general, I prefer to return an Optional.empty, instead of null. But I noticed that many (if not all) converters are already returning null in case of a failed conversion